numba/np/ufunc/_internal.c

```diff
@@ -210,7 +210,7 @@ dufunc_init(PyDUFuncObject *self, PyObject *args, PyObject *kws)
     /* Construct the UFunc. */
     tmp = PyObject_GetAttrString(py_func_obj, "__name__");
     if (tmp) {
-        name = PyString_AsString(tmp);
+        name = PyString_AsString(PyString_FromFormat("%s.ufunc", PyString_AsString(tmp)));
```

Comment:
We put in the change in numpy master.  The tricky part with the quick fix here is the memory management (assuming a dufunc can be deleted).  The string object currently created can never be dereferenced.

There is definitely much to be cleaned up, and it has to start in NumPy :/.  NumPy solves this by allocating the name as part of the single `ufunc->ptr`.  I expect numba already does enough dark magic that it could do that as well, but it may be tricky.  The other thing that looks a bit like it is a work-around is the `keepalive` list here?

Reply:
>  The string object currently created can never be dereferenced.

Hmm, are you saying that my PR makes a string that is never dereferenced? Or it was currently like that on numba? Or that the change to NumPy means that the string never gets dereferenced?

Reply:
Sorry, the first: this PR creates a string that is never dereferenced, even if the ufunc is deleted. But, the code is tricky, so I am not sure that it was even correct before.  And solving it seems not super straight forward.